### ISSUE TITLE
Automatically save PGN in engine v. engine game.

### ISSUE THREAD

**Issue by trumptonshire** [tmountain/uchess#5]
Request uchess feature to permit automatic saving of PGN output in engine v. engine games, similar to that of cutechess-cli.

For example:
`.\cutechess-cli -tournament gauntlet -concurrency 1 -pgnout lc0-stockfish-1.pgn -engine conf=lc0 -engine conf=stockfish -each proto=uci ponder tc=10+1 -openings file=openings-6ply-1000.pgn -repeat -recover -rounds 100 -games 2 | tee ./lc0-stockfish-1.txt`

During a engine v engine game, it is currently not possible to enter any commands into the interactive shell. Only pressing ENTER for the next move is permitted.

**Comment by tmountain** [comment]
This request makes sense. Would the ability to simple run the "save" command during engine v. engine games satisfy the use case?

**Comment by trumptonshire** [comment]
Yes please! Being able to save during an engine v engine game, either within the UI or from a command line parameter would be great!

**Comment by tmountain** [comment]
Cool, got it. I'll see about adding these changes during the next arc of work on this project.

**Comment by tmountain** [comment]
Engine-vs-engine games now auto-save their PGN, so you can watch a game play out and have the moves captured without any manual step.

Details:

1. Automatic save. In a CPU-vs-CPU game, when the game reaches an outcome (checkmate, stalemate, or draw) the PGN is written to uchess_<timestamp>.pgn in the current directory.

2. Save on quit. Because the shell accepts no commands during an engine-vs-engine game (only ENTER advances the moves), a long game you stopped early would previously be lost. Quitting an in-progress engine-vs-engine game now saves the PGN first, so nothing is thrown away.

3. Proper .pgn output. Saved games use a .pgn extension instead of .txt, so they open directly in Lichess, cutechess, and other analysis tools. The interactive "save" command writes .pgn as well.

Still to come, tracked with #4 (time control between two engines): a cutechess-style -pgnout <file> flag for choosing the output path, plus unattended play so engine-vs-engine games advance on a time control without pressing ENTER each move. Those two belong together, so I am handling them as one piece of work.

Thanks for the request and for the cutechess comparison, it was a helpful reference.
